tabulate.py

```diff
@@ -17,32 +20,65 @@
+  parser = argparse.ArgumentParser(description="")
+  parser.add_argument("-s", "--schema-file", dest="schema", metavar="PATH", default="schema.qdbd", required=False,
+                      help="Input schema for organizing data retrieved from input DICOM directories")
+  parser.add_argument("-d", "--input-directory", dest="inputDirectory", metavar="PATH", default="-", required=True,
+                      help="Input directory to recursively search and read DICOM information into tables")
+  parser.add_argument("-o", "--output-directory", dest="outputDirectory", metavar="PATH", default="-", required=True,
+                      help="Output directory to write tables in csv format to")
+  parser.add_argument("-dcmqi", "--dcmqi-path", dest="dcmqiPath", metavar="PATH",
+                      default=os.environ.get('DCMQI_PATH', None), required=False,
+                      help="Binary directory of dcmqi which is needed for reading DICOM SR TID1500")
+  args = parser.parse_args(argv)
+
+  if not args.dcmqiPath:
+    logging.warning("Parsing of DICOM SR TID 1500 won't be possible without specifying the location of your dcmqi "
+                    "executables. You can either specify dcmqi as an environment variable 'DCMQI_PATH' or as an "
+                    "additional parameter '-dcmqi <DCMQI binary path>'")
+
+  if not os.path.exists(args.outputDirectory):
+    create = raw_input('Output directory does not exist. Would you like to create it? [y/n]:\n')
+    if create.lower() in ["y", "yes", "true"]:
+      print "Creating directory %s" %args.outputDirectory
+      os.makedirs(args.outputDirectory)
+    else:
+      if not create.lower() in ["", "n", "no", "false"]:
+        print ("Entered letters %s unknown to this program." % create)
+      return
+
+  tablesParser = QDBDParser(args.schema)
   tablesRules = tablesParser.getTablesSchema()
 
-  for root,dirs,files in os.walk(sys.argv[2]):
+  for root,dirs,files in os.walk(args.inputDirectory):
     for f in files:
-
-
-      tables = {}
+      tables = dict()
       tables["Instance2File"] = []
       for t in tablesRules.keys():
         tables[t] = []
 
       dcmName = os.path.join(root,f)
 
       try:
-        dicomParser = SRCDParser(dcmName, tablesRules, tempPath=tempPath)
+        dicomParser = SRCDParser(dcmName, tablesRules, tempPath=tempPath, dcmqiPath=args.dcmqiPath)
       except:
         print("Failed to read as DICOM:"+dcmName)
         continue
 
       try:
         dicomParser.parse()
-      except:
-        print("Failed to parse: "+dcmName)
+      except (DCMQINotFoundError, KeyError) as exc:
+        print ("Failed to read DICOM %s" % dcmName)
+        if type(exc) is exceptions.KeyError:
+          print ("Missing key: %s " % exc)
+        else:
+          print ("Make sure that you specified dcmqi path either in your environment variable 'DCMQI_PATH' or as an "
+                "additional parameter '-dcmqi <DCMQI binary path>'")
+        print ("Skipping %s" %dcmName)
```

Comment:
I don't think it is a good idea to skip DICOM files when there is any of the attributes missing. They should be set to empty values, but parsing should continue for those files!

Reply:
I need to look into this before the PR is ready.